mobile/src/people/departments/departments-horizontal-scrollable-list.tsx

```diff
@@ -1,133 +1,104 @@
     scrollToEnd(): void;
 }
 
 export interface DepartmentsListStateDescriptor {
     currentPage: number;
-    // forceComponentRender: boolean;
-    // manualScrollMode: boolean;
 }
 
 export class DepartmentsHScrollableList extends Component<DepartmentsHScrollableListProps> {
-    private scrollView: ScrollViewComponent;
     private employeeCards: EmployeeCardWithAvatar[];
-    private animatedValue: Animated.Value;
+    private scrollView: ScrollViewComponent;
 
-    public shouldComponentUpdate(nextProps: DepartmentsHScrollableListProps) {	
-        if (this.props.departmentsLists !== nextProps.departmentsLists) {	
-            return true;	
-        } else {	
-            const employees = this.props.employees.employeesById.filter(this.props.employeesPredicate);	
-            const nextEmployees = nextProps.employees.employeesById.filter(nextProps.employeesPredicate);	
+    public componentDidMount() {
+        const x: number = this.props.departmentsLists ? this.props.departmentsLists.currentPage : 0;
+        const dep = this.props.departments[x];
+        if (dep) {
+            this.props.requestEmployeesForDepartment(dep.departmentId);
+        }
+        const curOffsetX = Dimensions.get('window').width * x;
 
-            return !employees.equals(nextEmployees);
-        }	
+        setTimeout(() => {
+            const view = this.scrollView as ScrollViewComponent;
+            if (view) {
+                view.scrollTo({y: 0, x: curOffsetX});
+            }
+        });
     }
 
     public render() {
+        const { headDepartment, departments } = this.props;
+        const subDepartments = headDepartment != null && departments.length > 0 ? departments : null;
         this.employeeCards = [];
-        
-        const { headDepartmentId, headDepartmentChiefId, departments } = this.props;
-        const subDepartments = headDepartmentId != null && departments.length > 0 ? departments : null;
-        let subordinates;
-        
-        if (this.props.employees.employeeIdsByDepartment.has(headDepartmentId) && this.props.employees.employeeIdsByDepartment.get(headDepartmentId).size > 0) {
-            subordinates = this.props.employees.employeesById.filter(this.props.employeesPredicate).toArray();
-        } else {
-            subordinates = null;
-        }
+        const cur = this;
 
         return <View>
             <ScrollView 
                 horizontal 
                 pagingEnabled
-                contentOffset={{x: Dimensions.get('window').width * (this.props.departmentsLists !== undefined ? this.props.departmentsLists.currentPage : departments.length), y: 0}}
                 showsHorizontalScrollIndicator={false}
-                onMomentumScrollEnd={this.onMomentumScrollEnd}
-                onScrollBeginDrag={this.onScrollBeginDrag}
-                ref={ref => this.scrollView = ref as ScrollViewComponent}
+                onMomentumScrollEnd={this.onMomentumScrollEnd.bind(this)}
```

Comment:
revert this change and have callbacks as `private onMomentumScrollEnd = (event: NativeSyntheticEvent<NativeScrollEvent>) => {` ? Lambdas already do this bind for you